src/gaussian.cc

```diff
@@ -109,9 +109,11 @@ namespace gauss2d
     const Gaussian & Gaussians::at_const(size_t i) const { return *(_data.at(i)); }
 
     typename Gaussians::Data::iterator Gaussians::begin() noexcept  {return _data.begin(); }
-    typename Gaussians::Data::const_iterator Gaussians::cbegin() const noexcept { return _data.begin(); }
+    typename Gaussians::Data::const_iterator Gaussians::begin() const noexcept { return _data.cbegin(); }
+    typename Gaussians::Data::const_iterator Gaussians::cbegin() const noexcept { return _data.cbegin(); }
```

Comment:
I'm confused here. Why do you have `Gaussians::begin` twice, once as an iterator and once as a constant iterator, and why do `Gaussians::begin` and `Gaussians::cbegin` return the same thing? I'm assuming that this has to do with the difference between using an `iterator` vs a `const_iterator`, but since they both point to the same object it will always be one or the other, right?

The same goes for `Gaussians::end` below.

Reply:
The short answer is that `begin` and `cbegin` are part of the standard [iterator interface](https://en.cppreference.com/w/cpp/iterator/begin). The overloaded `const_iterator` `begin/end` are there to support range-based for loop access like `for(const auto & elem : container)`, which call `begin` and `end` and will not compile if they only return a non-const interator.

More generally, I would hope there's a better way of writing very thin wrappers/extensions of STL containers, but I don't know how.